packages/synapse-react-client/src/components/widgets/Checkbox.tsx

```diff
@@ -1,17 +1,27 @@
+import Typography from '@mui/material/Typography'
 import { uniqueId as _uniqueId } from 'lodash-es'
 import React, { useState } from 'react'
 
-export type CheckboxProps = React.PropsWithChildren<{
-  label: string
-  hideLabel?: boolean
-  checked?: boolean
-  className?: string
-  onChange: (newValue: boolean) => void
-  isSelectAll?: boolean
-  onClick?: (event: React.SyntheticEvent<HTMLDivElement>) => void
-  disabled?: boolean
-  'data-testid'?: string
-}>
+export type CheckboxProps = React.PropsWithChildren<
+  {
+    hideLabel?: boolean
+    checked?: boolean
+    className?: string
+    onChange: (newValue: boolean) => void
+    isSelectAll?: boolean
+    onClick?: (event: React.SyntheticEvent<HTMLDivElement>) => void
+    disabled?: boolean
+    'data-testid'?: string
+    'aria-label'?: string
+  } & (
+    | {
+        label?: React.ReactNode
+        /* If the label is not a string, then we require an aria-label */
+        'aria-label': string
+      }
+    | { label: string }
+  )
```

Comment:
Add ability to add an arbitrary React component as a label, but if it's not a string, require an `aria-label`